Binary files arch/arm64/boot/dts/allwinner/overlay/sun50i-h616-fixup.scr and arch/arm64/boot/dts/allwinner/overlay/sun50i-h616-fixup.scr differ
--- arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi	2024-07-09 20:03:48.000000000 -0700.orig
+++ arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi
@@ -361,31 +361,13 @@
 			};
 
 			/omit-if-no-ref/
-			i2c0_pi_pins: i2c0-pi-pins {
-				pins = "PI5", "PI6";
-				function = "i2c0";
-			};
-
-			/omit-if-no-ref/
-			i2c1_ph_pins: i2c1-ph-pins {
-				pins = "PH0", "PH1";
-				function = "i2c1";
-			};
-
-			/omit-if-no-ref/
-			i2c1_pi_pins: i2c1-pi-pins {
+			i2c1_pins: i2c1-pins {
 				pins = "PI7", "PI8";
 				function = "i2c1";
 			};
 
 			/omit-if-no-ref/
-			i2c2_ph_pins: i2c2-ph-pins {
-				pins = "PH2", "PH3";
-				function = "i2c2";
-			};
-
-			/omit-if-no-ref/
-			i2c2_pi_pins: i2c2-pi-pins {
+			i2c2_pins: i2c2-pins {
 				pins = "PI9", "PI10";
 				function = "i2c2";
 			};
@@ -396,12 +378,6 @@
 				function = "i2c3";
 			};
 
-			i2c3_pa_pins: i2c3-pa-pins {
-				pins = "PA10", "PA11";
-				function = "i2c3";
-				bias-pull-up;
-			};
-
 			/omit-if-no-ref/
 			i2c4_ph_pins: i2c4-ph-pins {
 				pins = "PH6", "PH7";
@@ -825,21 +801,6 @@
 			status = "disabled";
 		};
 
-		i2c0: i2c@5002000 {
-			compatible = "allwinner,sun50i-h616-i2c",
-				     "allwinner,sun8i-v536-i2c",
-				     "allwinner,sun6i-a31-i2c";
-			reg = <0x05002000 0x400>;
-			interrupts = <GIC_SPI 6 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&ccu CLK_BUS_I2C0>;
-			resets = <&ccu RST_BUS_I2C0>;
-			pinctrl-names = "default";
-			pinctrl-0 = <&i2c0_pi_pins>;
-			status = "disabled";
-			#address-cells = <1>;
-			#size-cells = <0>;
-		};
-
 		i2c1: i2c@5002400 {
 			compatible = "allwinner,sun50i-h616-i2c",
 				     "allwinner,sun8i-v536-i2c",
@@ -849,7 +810,7 @@
 			clocks = <&ccu CLK_BUS_I2C1>;
 			resets = <&ccu RST_BUS_I2C1>;
 			pinctrl-names = "default";
-			pinctrl-0 = <&i2c1_ph_pins>;
+			pinctrl-0 = <&i2c1_pins>;
 			status = "disabled";
 			#address-cells = <1>;
 			#size-cells = <0>;
@@ -864,7 +825,7 @@
 			clocks = <&ccu CLK_BUS_I2C2>;
 			resets = <&ccu RST_BUS_I2C2>;
 			pinctrl-names = "default";
-			pinctrl-0 = <&i2c2_ph_pins>;
+			pinctrl-0 = <&i2c2_pins>;
 			status = "disabled";
 			#address-cells = <1>;
 			#size-cells = <0>;
@@ -919,14 +880,15 @@
 
 		spi1: spi@5011000 {
 			compatible = "allwinner,sun50i-h616-spi",
-				     "allwinner,sun8i-h3-spi";
+				     "allwinner,sun8i-h3-spi",
+                                     "spi-dev";
 			reg = <0x05011000 0x1000>;
 			interrupts = <GIC_SPI 13 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&ccu CLK_BUS_SPI1>, <&ccu CLK_SPI1>;
 			clock-names = "ahb", "mod";
 			resets = <&ccu RST_BUS_SPI1>;
 			pinctrl-names = "default";
-			pinctrl-0 = <&spi1_pins>, <&spi1_cs1_pin>;
+			pinctrl-0 = <&spi1_pins>, <&spi1_cs0_pin>, <&spi1_cs1_pin>;
 			dmas = <&dma 23>, <&dma 23>;
 			dma-names = "rx", "tx";
 			status = "disabled";
--- arch/arm64/boot/dts/allwinner/sun50i-h618-orangepi-zero2w.dts	2024-07-09 20:03:48.000000000 -0700.orig
+++ arch/arm64/boot/dts/allwinner/sun50i-h618-orangepi-zero2w.dts
@@ -282,48 +282,26 @@
 	status = "okay";
 };
 
-&i2c0 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&i2c0_pi_pins>;
-	status = "disabled";
-};
-
 &i2c1 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&i2c1_pi_pins>;
-	status = "disabled";
+	status = "okay";
 };
 
 &i2c2 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&i2c2_pi_pins>;
-	status = "disabled";
-};
-
-&i2c3 {
 	status = "okay";
-	pinctrl-names = "default";
-	pinctrl-0 = <&i2c3_pa_pins>;
+};
 
-	ac200_x: mfd@10 {
-		compatible = "x-powers,ac200";
-		reg = <0x10>;
-		clocks = <&ac200_pwm_clk>;
-		// ephy id
-		nvmem-cells = <&ephy_calibration>;
-		nvmem-cell-names = "calibration";
-
-		ac200_ephy: phy {
-			compatible = "x-powers,ac200-ephy";
-			status = "okay";
-		};
+&spi1 {
+	spidev@0 {
+		compatible = "spi-dev";
+		status = "okay";
+		reg = <0>;
 	};
-};
 
-&i2c4 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&i2c4_ph_pins>;
-	status = "disabled";
+	spidev@1 {
+		compatible = "spi-dev";
+		status = "okay";
+		reg = <1>;
+	};
 };
 
 &uart2 {
@@ -350,43 +328,6 @@
 	status = "disabled";
 };
 
-&spi0  {
-	status = "okay";
-	pinctrl-names = "default";
-	pinctrl-0 = <&spi0_pins>, <&spi0_cs0_pin>;
-
-	spidev@0 {
-		status = "disabled";
-		compatible = "rohm,dh2228fv";
-		reg = <0>;
-		spi-max-frequency = <1000000>;
-	};
-
-	flash@0 {
-		status = "okay";
-		#address-cells = <1>;
-		#size-cells = <1>;
-		compatible = "jedec,spi-nor";
-		reg = <0>;
-		spi-max-frequency = <40000000>;
-	};
-};
-
-&spi1 {
-	status = "disabled";
-	#address-cells = <1>;
-	#size-cells = <0>;
-	pinctrl-names = "default";
-	pinctrl-0 = <&spi1_pins>, <&spi1_cs1_pin>;
-
-	spidev@1 {
-		compatible = "rohm,dh2228fv";
-		status = "disabled";
-		reg = <1>;
-		spi-max-frequency = <1000000>;
-	};
-};
-
 &codec {
 	allwinner,audio-routing =
 	        "Line Out", "LINEOUT";
--- drivers/net/phy/Makefile	2024-07-09 20:03:48.000000000 -0700.orig
+++ drivers/net/phy/Makefile
@@ -2,7 +2,7 @@
 # Makefile for Linux PHY drivers
 
 libphy-y			:= phy.o phy-c45.o phy-core.o phy_device.o \
-				   linkmode.o sunxi-ephy.o
+				   linkmode.o
 mdio-bus-y			+= mdio_bus.o mdio_device.o
 
 ifdef CONFIG_MDIO_DEVICE
